File: extensions/twitter/src/main/java/app/revanced/twitter/patches/links/OpenLinksWithAppChooserPatch.java
package app.revanced.twitter.patches.links;

import android.content.Context;
import android.content.Intent;

import app.revanced.extension.shared.Logger;

@SuppressWarnings("unused")
@Deprecated(forRemoval = true)
public final class OpenLinksWithAppChooserPatch {

    /**
     * Injection point.
     */
    public static void openWithChooser(final Context context, final Intent intent) {
        Logger.printInfo(() -> "Opening intent with chooser: " + intent);

        intent.setAction("android.intent.action.VIEW");

        context.startActivity(Intent.createChooser(intent, null));
    }
}

File: patches/bin/main/app/revanced/patches/twitter/misc/links/OpenLinksWithAppChooserPatch.kt
package app.revanced.patches.twitter.misc.links

import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
import app.revanced.patcher.patch.bytecodePatch
import app.revanced.patches.twitter.misc.extension.sharedExtensionPatch

@Suppress("unused")
val openLinksWithAppChooserPatch = bytecodePatch(
    name = "Open links with app chooser",
    description = "Instead of opening links directly, open them with an app chooser. " +
        "As a result you can select a browser to open the link with.",
    use = false,
) {
    dependsOn(sharedExtensionPatch)

    compatibleWith("com.twitter.android"("10.48.0-release.0"))

    execute {
        val methodReference =
            "Lapp/revanced/extension/twitter/patches/links/OpenLinksWithAppChooserPatch;->" +
                "openWithChooser(Landroid/content/Context;Landroid/content/Intent;)V"

        openLinkFingerprint.method.addInstructions(
            0,
            """
                invoke-static { p0, p1 }, $methodReference
                return-void
            """,
        )
    }
}
